proxy/httpproxy/reverse.go

```diff
@@ -125,32 +125,38 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 		}()
 
 		defer func() {
-			completeCh <- true
+			close(completeCh)
 		}()
 	}
 
-	var res *http.Response
+	if proxybody != nil {
+		proxyreq.Body = ioutil.NopCloser(bytes.NewBuffer(proxybody))
+	}
 
+	var res *http.Response
 	for _, ep := range endpoints {
-		if proxybody != nil {
-			proxyreq.Body = ioutil.NopCloser(bytes.NewBuffer(proxybody))
-		}
 		redirectRequest(proxyreq, ep.URL)
 
-		res, err = p.transport.RoundTrip(proxyreq)
 		if atomic.LoadInt32(&requestClosed) == 1 {
 			return
 		}
+
+		res, err = p.transport.RoundTrip(proxyreq)
 		if err != nil {
 			reportRequestDropped(clientreq, failedSendingRequest)
 			plog.Printf("failed to direct request to %s: %v", ep.URL.String(), err)
 			ep.Failed()
+			res = nil
```

Comment:
potentially `reportRequestDropped` will be updated twice

Reply:
This metric corresponds to the number of failures of proxy to backend. If have 5 back ends,  the metric will increase by 5 in the case of failure